### DISCUSSION TITLE
Costume_Editor:Mega_Upgrade

### DISCUSSION THREAD

**Discussion by fierin** [ScratchAddons/ScratchAddons#3122] [Ideas]
### Immediate
This could be merged into Onion Skinning.

### Description
This is somewhat a MegaThread for Costume editor features.
Feel free to add more to the Idea!

#### Animate In-Editor
Allow playing/loop through the costumes with a button and or a keybind.
If possible add a setting on how much is the FPS of the animation.
#### Rigging
This may be a bit much for a chrome addon… but who thought that any of the features SA has already featured was?
Allow rigging. Give a group of vectors a bone that connects to more bones which can be dragged to easily animate a character as a rig.
#### Interpolation
Staying on the topic of probably too ridiculous to add…
Interpolation; using rigs when animation you could get the frames to interpolate between themselves. This way you don't need to fill the gaps to make them smoother manually.

## Disclosure

This is a PURELY Experimental idea and is prone to change… additions… and turning down of some features. I would like to see what you think!

**Comment by DroneBetter** [comment]
Can rigging not be achieved by having an image per 'bone' and rotating each about its origin from its parent's end?

**Comment by rimopa** [comment]
Center horizontally and center vertically buttons in vector
